crun.1

```diff
@@ -711,6 +711,58 @@ If the \fB\fCidmap\fR option is specified then the mount is ID mapped using the
 target user namespace.  This is an experimental feature and can change at any time
 without notice.
 
+.PP
+The \fB\fCidmap\fR option supports a custom mapping that can be different
+than the user namespace used by the container.
+
+.PP
+The mapping can be specified after the \fB\fCidmap\fR option like:
+\fB\fCidmap=uids=0-1-10;gids=0-100-10\fR\&.
+
+.PP
+The only two options that are currently supported are \fB\fCuids\fR and \fB\fCgids\fR\&.
+
+.PP
+When a custom mapping is specified, a new user namespace is created
+for the idmapped mount.
+
+.PP
+If no option is specified, then the container user namespace is used.
+
+.PP
+If the specified mapping is prepended with a '+' then the mapping is
+considered relative to the container user namespace.  The host ID for
+the mapping is changed to account for the relative position of the
+container user in the container user namespace.
+
+.PP
+For example, the mapping: \fB\fCuids=+1-1-10\fR, given a configuration like
+
+.PP
+.RS
+
+.nf
+"uidMappings": [
+      {
+        "containerID": 0,
+        "hostID": 1,
+        "size": 1000
+      }
+    ]
+
+.fi
+.RE
+
+.PP
+will be converted to the absolute value \fB\fCuids=1-3-10\fR, where 3 is
+calculated by adding 1 (container ID in the uids mapping) + 2 (the
+host ID for the UID 1 from the relative mapping).
```

Comment:
Wouldn't it be better if `uids=+1-1-10` would result in `uids=1-2-10` instead? So that value[0] would not influence value[1]:
GetHostUID({"containerID": 0,"hostID": 1,"size": 1000}, 1) = 2
value[1] += mappings[i]->host_id + (~~value[0]~~ - mappings[i]->container_id);

If value[0] would influence value[1] as it does now I would quite often have to use negative values for value[1] which also seems a bit error-prone since the separator is already  `-`.
The same argument applies to the prefix   `+` to indicate a relative mapping. Maybe replace it with `@` or `R`.

---
With these suggestion applied:

```
[lukas@PC6061B feb16]$ stat mnt/
  File: mnt/
  Size: 0               Blocks: 0          IO Block: 4096   directory
Device: 0,46    Inode: 10914100    Links: 1
Access: (0755/drwxr-xr-x)  Uid: ( 1000/   lukas)   Gid: ( 1000/   lukas)
[lukas@PC6061B feb16]$ sudo podman run --mount "type=bind,src=$(pwd)/mnt,dst=/mnt,idmap=uids=@1000-1000-100;gids=@1000-1000-100" --userns=auto --rm fedora stat -c '%u:%g' /mnt
1000:1000
[lukas@PC6061B feb16]$ sudo podman run --mount "type=bind,src=$(pwd)/mnt,dst=/mnt,idmap=uids=@1000-0-100;gids=@1000-0-100" --userns=auto --rm fedora stat -c '%u:%g' /mnt
0:0
```

Reply:
I've amended your suggestions in the new version.

Can you try if it is fine now?

Reply:
> I am fine to do these changes. So you are suggesting we use `value[1] += mappings[i]->host_id - mappings[i]->container_id;`?

Yes and replacing the prefix `+` for a relative mapping with something more intuitive like `@`.

Reply:
>Can you try if it is fine now?

Thanks now it works. 

I am wondering it would be good for consistency with  --uidmap/gidmap (container id comes before that host) to swap value[1] and value[0].

This would also require swapping them when writting the mapping:
sprintf (it_mappings, "%ld %ld %ld\n", value[1], value[0], value[2]);
And this swap would also apply if the mapping is not relative.

Either way, if you make this change or not I would recommend explaining more in the documentation on which value comes first (on_disk_uid or (container/host)_uid).

Reply:
thanks for the suggestion.  I've used the wrong code comment, but `value[0]` refers to the container ID while `value[1]` is the host ID, in the same order as they are written to the `[ug]id_map` files.

Pushed a new version